autoload/ale/completion.vim

```diff
@@ -321,13 +324,21 @@ function! ale#completion#ParseTSServerCompletionEntryDetails(response) abort
         endif
 
         " See :help complete-items
-        call add(l:results, {
+        let l:result = {
         \   'word': l:suggestion.name,
         \   'kind': l:kind,
         \   'icase': 1,
         \   'menu': join(l:displayParts, ''),
         \   'info': join(l:documentationParts, ''),
-        \})
+        \}
+
+        if has_key(l:suggestion, 'codeActions')
+            let l:result.user_data = json_encode({
```

Comment:
Try removing `json_encode` here and `json_decode` below. I think you can store `Dictionary` values too.

Reply:
I thought so too, but I get `E731: using Dictionary as a String` when I remove the `json_encode`/`json_decode` functions :(